Tests/NIOWebSocketTests/NIOWebSocketFrameAggregatorTests.swift

```diff
@@ -17,53 +17,35 @@ import NIO
 import NIOWebSocket
 
 final class NIOWebSocketFrameAggregatorTests: XCTestCase {
-    var channel: EmbeddedChannel!
+    let channel = EmbeddedChannel(
```

Comment:
This is actually a problem for weird `XCTestCase` reasons. `EmbeddedChannel` wants you to call `finish` when you're done with it. For some weird reasons however each `XCTestCase` gets initialised _twice_ (and your code only runs on one of those). Hence, if you have a `let something: ...` it'll be initialised but you can't finish its lifecycle.

I'd recommend going for `var channel: EmbeddedChannel!`, configuring it in `setUp` and tearing it down in `tearDown`.

@glbrntt I got this right with the `XCTestCase`, right?

Reply:
Yes that sounds right but IIRC this only happens if you select a single test to run in Xcode, running all tests is fine.

Reply:
Thanks, didn't know that! But does EmbeddedChannel requires explicit shutdown even if if was never used? I thought that  only `MultithreadedEventLoopGroup` does.